chipsec/helper/linux/linuxhelper.py

```diff
@@ -1042,124 +1043,124 @@ def EFIVARS_list_EFI_variables(self):
-        in_buf = struct.pack('<11' + self._pack, rcx, rdx, r8, r9, r10, r11, rax, rbx, rdi, rsi, xmm_buffer)
+    def hypercall(self, rcx: int, rdx: int, r8: int, r9: int, r10: int, r11: int, rax: int, rbx: int, rdi: int, rsi: int, xmm_buffer: int) -> int:
+        in_buf = struct.pack(f'<11{self._pack}', rcx, rdx, r8, r9, r10, r11, rax, rbx, rdi, rsi, xmm_buffer)
         out_buf = self.ioctl(IOCTL_HYPERCALL, in_buf)
-        return struct.unpack('<11' + self._pack, out_buf)[0]
+        return struct.unpack(f'<11{self._pack}', out_buf)[0]
 
     #
     # Interrupts
     #
-    def send_sw_smi(self, cpu_thread_id, SMI_code_data, _rax, _rbx, _rcx, _rdx, _rsi, _rdi):
+    def send_sw_smi(self, cpu_thread_id: int, SMI_code_data: int, _rax: int, _rbx: int, _rcx: int, _rdx: int, _rsi: int, _rdi: int) -> Optional[int]:
         self.set_affinity(cpu_thread_id)
-        in_buf = struct.pack("7" + self._pack, SMI_code_data, _rax, _rbx, _rcx, _rdx, _rsi, _rdi)
+        in_buf = struct.pack(f'7{self._pack}', SMI_code_data, _rax, _rbx, _rcx, _rdx, _rsi, _rdi)
         out_buf = self.ioctl(IOCTL_SWSMI, in_buf)
-        ret = struct.unpack("7" + self._pack, out_buf)
+        ret = struct.unpack(f'7{self._pack}', out_buf)[0]
         return ret
 
     #
     # File system
     #
-    def get_tool_info(self, tool_type):
+    def get_tool_info(self, tool_type: str) -> Tuple[Optional[str], str]:
         tool_name = _tools[tool_type] if tool_type in _tools else None
         tool_path = os.path.join(get_tools_path(), self.os_system.lower())
         return tool_name, tool_path
 
-    def getcwd(self):
+    def getcwd(self) -> str:
         return os.getcwd()
 
-    def get_page_is_ram(self):
+    def get_page_is_ram(self) -> Optional[bytes]:
         PROC_KALLSYMS = "/proc/kallsyms"
         symarr = chipsec.file.read_file(PROC_KALLSYMS).splitlines()
         for line in symarr:
-            if "page_is_ram" in line:
-                return line.split(" ")[0]
+            if b"page_is_ram" in line:
+                return line.split(b" ")[0]
+        return None
 
-    def get_phys_mem_access_prot(self):
+    def get_phys_mem_access_prot(self) -> Optional[bytes]:
         PROC_KALLSYMS = "/proc/kallsyms"
         symarr = chipsec.file.read_file(PROC_KALLSYMS).splitlines()
         for line in symarr:
-            if "phys_mem_access_prot" in line:
-                return line.split(" ")[0]
+            if b"phys_mem_access_prot" in line:
+                return line.split(b" ")[0]
+        return None
 
     #
     # Logical CPU count
     #
-    def get_threads_count(self):
+    def get_threads_count(self) -> int:
         import multiprocessing
         return multiprocessing.cpu_count()
 
     #
     # Speculation control
     #
-    def retpoline_enabled(self):
-        raise UnimplementedAPIError("retpoline_enabled")
+    def retpoline_enabled(self) -> bool:
+        return False
```

Comment:
This should stay as unimplemented. In UEFI Shell, I can see this just returning False. But in Linux it is possible to have it, we just haven't implemented the check for it.